contracts/transfer/circuits/src/gadgets/secret_key.rs

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) DUSK NETWORK. All rights reserved.
+// Licensed under the MPL 2.0 license. See LICENSE file in the project root for details.
+
+use dusk_plonk::constraint_system::ecc::scalar_mul::fixed_base::scalar_mul;
+use dusk_plonk::jubjub::{
+    AffinePoint, GENERATOR_EXTENDED, GENERATOR_NUMS_EXTENDED,
+};
+use dusk_plonk::prelude::*;
+
+use plonk_gadgets::AllocatedScalar;
+
+// Prove that the amount inputted equals the amount outputted
+pub fn sk_knowledge(composer: &mut StandardComposer, sk: AllocatedScalar, pk: AffinePoint) {
+    
+    let p1 = scalar_mul(composer, sk.var, GENERATOR_EXTENDED);
+
+    composer.assert_equal_public_point(*p1.point(), pk);
+    
+}
```

Comment:
Dunno if this is even needed. Makes more sense than merkle or range fn's but its still nothig.